### ISSUE TITLE
I have new issue with AdapterDateFnsJalali in LocalizationProvider !!

### ISSUE THREAD

**Issue by isrezaei** [mui/mui-x#9147]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

Steps:
1.https://1eib5z.csb.app/   ==> i use AdapterDateFnsJalali 
2.https://k08dq5.csb.app/  ==> i use AdapterDayjs

### Current behavior 😯

when i use dateAdapter={AdapterDateFnsJalali} in LocalizationProvider , in slotProps={{day : (ownerState)}},
data of ownerState , part day object , i got a full data time string !
but when i use AdapterDayjs in dateAdapter , in slotProps={{day : (ownerState)}} , ownerState part of day , i got information about every day month and years ! why ?

in this case i got full string time stamp :

    <LocalizationProvider
        dateAdapter={AdapterDateFnsJalali} // i use AdapterDateFnsJalali
        adapterLocale={faJalali}
        localeText={faIR.components.MuiLocalizationProvider.defaultProps.localeText}  >
        <DemoContainer components={['DatePicker']}>
            <DatePicker
                label="date"
                clearable
                slotProps={{
                    day: (ownerState) => {
                        const DATES = {
                            month : ownerState.day['$M'], //i got string full time stamp !!
                            day : ownerState.day['$D'], //i got string full time stamp !!
                        }
                    }
                }}
            />
        </DemoContainer>
    </LocalizationProvider>
but in this case i got full info about date

    <LocalizationProvider
        dateAdapter={AdapterDayjs} //i use AdapterDayjs
    >
        <DemoContainer components={['DatePicker']}>
            <DatePicker
                label="date"
                clearable
                slotProps={{
                    day: (ownerState) => {
                        const DATES = {
                            month : ownerState.day['$M'], i got courrect info about each month 
                            day : ownerState.day['$D'],i got courrect info about each day
                        }
                    }
                }}
            />
        </DemoContainer>
    </LocalizationProvider>

### Expected behavior 🤔

should i got information about date like AdapterDayjs ! 

    <LocalizationProvider
        dateAdapter={AdapterDayjs}  //i use AdapterDayjs
    >
        <DemoContainer components={['DatePicker']}>
            <DatePicker
                label="date"
                clearable
                slotProps={{
                    day: (ownerState) => {
                        const DATES = {
                            month : ownerState.day['$M'],  i got courrect info about each month 
                            day : ownerState.day['$D'] ,i got courrect info about each day
                        }
                    }
                }}
            />
        </DemoContainer>
    </LocalizationProvider>

**Comment by isrezaei** [comment]
This information does not give me the numbers of the day! Based on the IP that I have, I want to compare the number of each day with the number of each day in the material calendar and change it to a specific day if necessary, but this current information from the calendar, the day section, gives me, for example, It gives the third day of the week, while I want the numbers of the days like 1 2 3 4 and... 31!

This question arises from here:
[https://github.com/mui/mui-x/issues/9091](url)

**Comment by alexfauquette** [comment]
> but this current information from the calendar, the day section "day: ownerState.day.getDay()", for example, It gives the third day of the week

Yes, I took the first method I thaught about. I let you implement the exact behavior you want

> While I want the numbers of each days like 1 2 3 4 ... 31!
also i want my data be jalali format !

As said earlier, those are out of the scope of mui components, but about date-fns-jalali usage (and by extension date-fns). If you are blocked, you should either ask on StackOverflow, or have a look to the documentation of the object you are dealing with:

- https://www.npmjs.com/package/date-fns-jalali
- https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate
- https://date-fns.org/docs/Getting-Started/
